src/analysis/processing/qgsalgorithmsaveselectedfeatures.cpp

```diff
@@ -29,7 +29,7 @@ void QgsSaveSelectedFeatures::initAlgorithm( const QVariantMap & )
 {
   addParameter( new QgsProcessingParameterVectorLayer( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ),
```

Comment:
Not related to this PR, but I'm wondering why we use `QgsProcessingParameterVectorLayer` here instead of feature source which is more flexible?

Reply:
Looks like it was directly adapted from the old QGIS2 code, and that change was overlooked

Reply:
It's because it works directly with QgsVectorLayer methods -- i.e. directly accesses the selected features from the layer. Feature source parameters themselves have no knowledge of selections, they just get a bunch of features and it's the parameter value which determines whether these are the selected features only.